### PR TITLE
rainloop service: init

### PR DESCRIPTION
**Pull request by Twey**
This pull request introduces a service for running (multiple instances of) the `rainloop` webmail client.

It sits atop #97271.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/mail/rainloop.nix [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+  config = let
+    c = mapAttrsToList instance cfg;
+  in mkIf (cfg != { }) {
+    system.activationScripts.rainloop = {
+      deps = [];
+      text = ''
+        ${pkgs.coreutils}/bin/mkdir -p /var/lib/rainloop
+        ${pkgs.coreutils}/bin/chmod 755 /var/lib/rainloop
+      '';
```
Comment (doronbehar) [inline nixos/modules/services/mail/rainloop.nix:103]: How about `systemd.tmpfiles.rules`?

File: nixos/modules/services/mail/rainloop.nix [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      phpEnv."PATH" = lib.makeBinPath [ php-package ];
+      phpOptions = ''
+        upload_max_filesize = 40m
+        post_max_size = 49M
+      '';
+    };
+
+    vhost.name = domain;
+    vhost.value = {
+      extraConfig = ''client_max_body_size 50m;'';
```
Comment (doronbehar) [inline nixos/modules/services/mail/rainloop.nix:43]:
```suggestion
      extraConfig = ''
        client_max_body_size 50m;
      '';
```

**Review by doronbehar — COMMENTED** [review]
Haven't tested, but diff looks good.

File: nixos/modules/services/mail/rainloop.nix (hunk 2 above)
Comment (Twey) [reply nixos/modules/services/mail/rainloop.nix:43]: What do the extra linebreaks do?

Comment (doronbehar) [reply nixos/modules/services/mail/rainloop.nix:43]: Nothing, but they make it more readable, IMO and future diffs look better.

**Comment by doronbehar** [comment]
Thank you for your contribution @Twey. I don't know a thing about PHP and nginx related services in Nixpkgs but I'd like to help you, and the changes look good. Considering also https://github.com/NixOS/nixpkgs/pull/97271, perhaps you could write a nixos test that will test the features you'd like to add? It's common to do so when adding new services.

**Comment by doronbehar** [comment]
Closing due to OP not responding.

**Pull request closed without merging**